docs/capi-vsphere.md

```diff
@@ -0,0 +1,178 @@
+VSPHERE_RESOURCE_POOL: "*/Resources"                          # The vSphere resource pool for your VMs
+VSPHERE_FOLDER: "vm"                                          # The VM folder for your VMs. Set to "" to use the root vSphere folder
+VSPHERE_TEMPLATE: "ubuntu-1804-kube-v1.17.3"                  # The VM template to use for your management cluster.
+CONTROL_PLANE_ENDPOINT_IP: "192.168.9.230"                    # the IP that kube-vip is going to use as a control plane endpoint
+VIP_NETWORK_INTERFACE: "ens192"                               # The interface that kube-vip should apply the IP to. Omit to tell kube-vip to autodetect the interface.
+VSPHERE_TLS_THUMBPRINT: "..."                                 # sha1 thumbprint of the vcenter certificate: openssl x509 -sha1 -fingerprint -in ca.crt -noout
+EXP_CLUSTER_RESOURCE_SET: "true"                              # This enables the ClusterResourceSet feature that we are using to deploy CSI
+VSPHERE_SSH_AUTHORIZED_KEY: "ssh-rsa AAAAB3N..."              # The public ssh authorized key on all machines in this cluster.
+                                                              #  Set to "" if you don't want to enable SSH, or are using another solution.
+VSPHERE_STORAGE_POLICY: ""                                    # This is the vSphere storage policy. Set it to "" if you don't want to use a storage policy.
+"CPI_IMAGE_K8S_VERSION": "v1.25.0"                            # The version of the vSphere CPI image to be used by the CPI workloads
+                                                              #  Keep this close to the minimum Kubernetes version of the cluster being created.
+K0S_VERSION: "v1.27.2+k0s.0"
+K0S_CP_VERSION: "v1.27.2-k0s.0"
+CSI_INSECURE: "1"
+NODE_IPAM_POOL_NAME: "ipam-ip-pool"
+NODE_IPAM_POOL_API_GROUP: "ipam.cluster.x-k8s.io"
+NODE_IPAM_POOL_KIND: "InClusterIPPool"
+NAMESERVER: "8.8.8.8"
+providers:
+  - name: incluster
+    url: https://github.com/kubernetes-sigs/cluster-api-ipam-provider-in-cluster/releases/latest/ipam-components.yaml
+    type: IPAMProvider
+
+```
+### Management cluster initialization
+
+To initialize the management cluster with vSphere infrastrcture provider and additional IPAM provider you can run:
+
+```
+clusterctl init --infrastructure vsphere --ipam incluster
+```
+
+For more details on Cluster API Provider vSphere see it's [docs](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/tree/main/docs).
+
+### IPAM IP pool creation
+
+1. Define IP pool that will be used to assign IP addresses to a child cluster machines in `capi-ipam.yaml`:
+
+```yaml
+apiVersion: ipam.cluster.x-k8s.io/v1alpha1
+kind: InClusterIPPool
+metadata:
+  name: ipam-ip-pool
+spec:
+  subnet: 192.168.117.0/24
+  gateway: 192.168.117.1
+  start: 192.168.117.152
+  end: 192.168.117.180
+```
+
+2. Deploy the IPAM configuration: 
+```
+kubectl apply -f capi-ipam.yaml
+```
+
+## MetalLB as Load Balancer solution
```

Comment:
This only applies to management clusters that are running in vSphere. Please add this as _optional_ for the ones with mgmt cluster on vSphere.